recipes/cpp_project_framework/all/conanfile.py

```diff
@@ -0,0 +1,99 @@
+from conan import ConanFile
+from conans import CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class CppProjectFrameworkConan(ConanFile):
+    name = "cpp_project_framework"
+    license = "GNU Affero General Public License v3.0"
+    homepage = "https://github.com/sheepgrass/cpp_project_framework"
+    url = "https://github.com/conan-io/conan-center-index"  # Package recipe repository url here, for issues about the package
+    description = "C++ Project Framework is a framework for creating C++ project."
+    topics = ("c++", "project", "framework")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+    }
+    generators = "cmake"
+    short_paths = True
+    exports_sources = "%s/*" % name, "test_package/*.*"
+    build_requires = "gtest/1.10.0", "doxygen/1.8.20", "benchmark/1.5.1"
+    exports_resources = ".gitignore", "LICENSE", "conanfile.txt", "CMakeLists.txt", "make.bat", "Makefile", "cpp_project_framework_callables.cmake", "cpp_project_framework.cmake"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16",
+            "gcc": "7",
+            "clang": "6",
+            "apple-clang": "10",
+        }
+
+    def validate(self):
+        if self.settings.os != "Linux" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration(f"{self.name} is just supported for Linux and Windows")
+
+        compiler = self.settings.compiler
+
+        if compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+
+        if compiler == "gcc" or compiler == "clang":
+            if compiler.get_safe("libcxx") != "libstdc++":
+                raise ConanInvalidConfiguration(f"only supported {compiler} with libstdc++")
+
+        min_version = self._minimum_compilers_version.get(str(compiler))
+        if not min_version:
+            self.output.warn(f"{self.name} recipe lacks information about the {compiler} compiler support.")
+        else:
+            if tools.Version(compiler.version) < min_version:
+                raise ConanInvalidConfiguration(f"{self.name} requires C++{self._minimum_cpp_standard} support. The current compiler {compiler} {compiler.version} does not support it.")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
```

Comment:
I am not at all familiar with this project but the fact it includes Conan gives it my vote 😉 

Why does this need to be build?

Reply:
thanks for your vote 😄~
I really love how Conan works and so I give my vote to Conan and integrate it into my framework~
Thanks for your review.  The build section is actually no need and I removed it in b99dee54c00499301e1a193101533ea23bd2bd06